pkg/query/selection.go

```diff
@@ -175,13 +201,22 @@ func (c Criterion) Validate() error {
 		return &util.UnsupportedQueryError{Message: fmt.Sprintf("separator %s is not allowed in %s with left operand \"%s\".", Separator, c.Type, c.LeftOp)}
 	}
 	for _, op := range c.RightOp {
-		if strings.ContainsRune(op, '\n') {
+		if strings.ContainsRune(op, '\n') && c.Type != ExistQuery {
 			return &util.UnsupportedQueryError{Message: fmt.Sprintf("%s with key \"%s\" has value \"%s\" contaning forbidden new line character", c.Type, c.LeftOp, op)}
 		}
 	}
 	return nil
 }
 
+func (c *Criterion) addTemplateParams(templateParameters TemplateParameters) {
```

Comment:
does anyone need this "add" functionality? why not just use assignment of the parameters field?

Reply:
Ok, I've removed the "add" functionality and simply set with the given parameters field.